### ISSUE TITLE
Format field in card view

### ISSUE THREAD

**Issue by m4dz** [cozy/cozy-contacts#63]
When view card, some fields may be formatted for display, like phone number (see screenshot).
![screen shot 2015-02-09 at 10 19 10](https://cloud.githubusercontent.com/assets/471486/6103515/82c34e62-b045-11e4-8e8d-9231f5855920.png)

E.g.: I often saved phone numbers w/o spaces (+33212345678), but the display render should format them to improve readability (+33 2 12 34 56 78).

**Comment by aenario** [comment]
That's pretty hard to do in a internationally compatible way. The google librairy for phone numbers might prove usefull.

**Comment by m4dz** [comment]
Yep. Maybe we could preset settings with user locale settings, and let him override those settings with a simple pattern (e.g. `+## # ## ## ## ##` > `+## ### ## ## ##`) ?

**Comment by hansbkk** [comment]
I ALWAYS format my phone numbers exactly as I prefer them to appear, which for the past copule of decades has been spaces rather than any parens or dashes.

If there is an app or service that doesn't like that formatting - e.g. doesn't work properly with phone numbers containing spaces - then I wouldn't use that app or service.

Have you (has anyone reading this) actually come across any problems in this area with vCards or CardDAV?

@m4dz The number of digits in the different components varies from one place to the next, or even depending on location or number type within a given country.

**Comment by m4dz** [comment]
@hansbkk, maybe you take the time to format the phone numbers exactly as you want, but this is _your_ choice, not the one of many users. When I add a contact from my call-log on my phone, I really don't care how it saves information for me. But I prefer to have it displayed in a consistent manner on _all_ the card of my addressbook.

Yes, I know that phone numbers are located, due to user location, preferences, and the phone number origin itself. It's not something that I forgot in my previous post, and I speak as a UX designer to mention this feature.

If I mention this, this is because:
- it's a sweet improvement but _not_ a critical missing feature, and this issue is here to remember it;
- this choice was already debates in others softwares, and (guess what?) it works:
![mac os 10.10 addressbook help: format numbers](https://cloud.githubusercontent.com/assets/471486/6144591/98bb0b7a-b1d9-11e4-9712-e4505b092142.png)

Note that my suggestion let yo choose your format, for one card or for all, just to let you display them the way you want…

**Comment by hansbkk** [comment]
Sure it's a great idea, and now I can see the need for it for those that allow/want contacts to be added by automated processes (not me for sure!)

However it seems like a big challenge to me to come up with A a rule engine with enough flexibility to handle the dozens if not hundreds of different "break patterns" and B a way for the user to express their preferences based on ? address field country string ? or  ? country code + area code ? because just doing it for the number of digits for sure isn't going to work.

Just in Thailand ("when beginning with +44") or Cambodia ("when beginning with +855") the user may want different strings for different international carriers (001,004,005,006,007,008 and 009). Mobile numbers have a different number of digits from landlines, large-city numbers have a smaller number of digits in the area code compared to provincial areas, the local part of the number can have six digits or seven. Biggest issue is that within the country the dialing string must start with a "0", which gets dropped when dialing in from outside. . .

And then I for one often use "after connection" strings, complete with pauses, to access a certain department or an individual's extension.
